src/pages/en/guides/rss.md

```diff
@@ -6,11 +6,139 @@ description: An intro to RSS in Astro
 
-You can create an RSS feed from any Astro page that uses a `getStaticPaths()` function for routing. Only dynamic routes can use `getStaticPaths()` today (see [Routing](/en/core-concepts/routing)).
+## Using `@astrojs/rss` (recommended)
 
-> We hope to make this feature available to all other pages before v1.0. As a workaround, you can convert a static route to a dynamic route that only generates a single page. See [Routing](/en/core-concepts/routing) for more information about dynamic routes.
+The `@astrojs/rss` package provides helpers for generating RSS feeds using [API endpoints](/en/core-concepts/astro-pages/#non-html-pages). This unlocks both static builds _and_ on-demand generation when using an [SSR adapter](/en/guides/server-side-rendering/#enabling-ssr-in-your-project).
 
-Create an RSS Feed by calling the `rss()` function that is passed as an argument to `getStaticPaths()`. This will create an `rss.xml` file in your final build based on the data that you provide using the `items` array.
+First, install `@astrojs/rss` using your preferred package manager:
+
+```bash
+# npm
+npm i @astrojs/rss
+# yarn
+yarn add @astrojs/rss
+# pnpm
+pnpm i @astrojs/rss
+```
+
+To generate your first RSS feed, create an `rss.xml.js` file under your `src/pages/` directory. `rss.xml` will be the output URL, so feel free to rename this if you prefer.
+
+Now, import the `rss` helper from the `@astrojs/rss` package. We will explore each of the parameters below:
+
+```js
+// src/pages/rss.xml.js
+import rss from '@astrojs/rss';
+
+export const get = () => rss({
+    // `<title>` field in output xml
+    title: 'Buzz’s Blog',
+    // `<description>` field in output xml
+    description: 'A humble Astronaut’s guide to the stars',
+    // list of `<item>`s in output xml
+    // simple example: generate items for every md file in /src/pages
+    // see "Generating items" section for required frontmatter and advanced use cases
+    items: import.meta.glob('./**/*.md'),
+    // (optional) inject custom xml
+    customData: `<language>en-us</language>`,
+  });
+```
+
+### Generating `items`
+
+The `items` field accepts either:
+1. [A list of RSS feed objects](#1-list-of-rss-feed-objects), each with a `link`, `title`, `pubDate`, and optional `description` and `customData` fields.
+2. [An `import.meta.glob(...)` result](#2-importmetaglob-result). **Only use this for `.md` files in the `src/pages/` directory!**
+
+#### 1. List of RSS feed objects
+
+We recommend this option for `.md` files outside of the `pages` directory. This is common when generating routes [via `getStaticPaths`](/en/reference/api-reference/#getstaticpaths). 
+
+For instance, say our blog posts are stored under a `src/posts/` directory. Each post has a `title`, `pubDate`, and `slug` in its frontmatter, where `slug` corresponds to the output URL on your site. We can generate an RSS feed using [Vite's `import.meta.globEager` helper](https://vitejs.dev/guide/features.html#glob-import) like so:
```

Comment:
Maybe instead of the conversational "say our blog posts ..."  I want to see something like:

Example: `.md` posts are located in `src/posts/`

And, I'm not *sure* we need to spell out what the frontmatter values are gonna be? If `slug` is particularly particular, then we can call that out. I know I'm rarely in the "this is overexplaining" camp, but it's a little visually distracting with all the inline code. Would a code example of the yaml front matter be helpful? Could we get away without it entirely, and assume that the corresponding `.md` files are going to have those properties in frontmatter?

Just thinkin' here.

Reply:
Hm, I think calling out the frontmatter is pretty necessary since `slug` is worth explaining. I personally find it less distracting than a full-blown `yaml` code block. Idk, I vote keeping that bit 🤷

Reply:
Happy to defer on that!

Reply:
Sounds good. Made that `.md` post suggestion though!